### PR TITLE
removing System.Text.Json dependency and instead using Azure.Core.Serialization.ObjectSerializer

### LINKED ISSUE #83: Change JSON serialization to use ObjectSerializer
See example here -- we should use `ObjectSerializer` from the Azure SDK to abstract away serialization technology. This likely should happen for GA because it could impact public API and slightly change behavior.

See here for an example: https://github.com/Azure/azure-sdk-for-net/blob/ef4430baed/sdk/core/Microsoft.Azure.Core.NewtonsoftJson/README.md. We'd likely add a `Serializer` property to our Options.

### PR DESCRIPTION
**Pull request by brettsam**
Fixes #83

### CODE REVIEW

File: src/DotNetWorker/Configuration/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -68,6 +68,11 @@ public static IFunctionsWorkerApplicationBuilder AddFunctionsWorker(this IServic
                     config.Bind(arguments);
                 });
 
+            if (configure != null)
+            {
+                services.Configure(configure);
```
Comment (fabiocav) [inline src/DotNetWorker/Configuration/ServiceCollectionExtensions.cs:73]: nit: `services?.Configure(configure)`

File: src/DotNetWorker/Configuration/WorkerOptions.cs [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
-namespace Microsoft.Azure.Functions.Worker.Configuration
+namespace Microsoft.Azure.Functions.Worker
 {
-    // TODO: Slim this down.
     public class WorkerOptions
     {
         public WorkerOptions()
         {
-            Capabilities = new List<string>();
+            var serializerOptions = new JsonSerializerOptions
```
Comment (fabiocav) [inline src/DotNetWorker/Configuration/WorkerOptions.cs:13]: Is there a reason we're setting this? Expectations from our internal protocol?

File: src/DotNetWorker/Converters/JsonPocoConverter.cs [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
     internal class JsonPocoConverter : IConverter
     {
-        private readonly IOptions<JsonSerializerOptions> _options;
+        private readonly ObjectSerializer _serializer;
 
-        public JsonPocoConverter(IOptions<JsonSerializerOptions> options)
+        public JsonPocoConverter(IOptions<WorkerOptions> options)
         {
-            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _serializer = options?.Value?.Serializer ?? throw new ArgumentNullException(nameof(options));
```
Comment (fabiocav) [inline src/DotNetWorker/Converters/JsonPocoConverter.cs:20]: The exception only applies if the parameter is null, for the properties, we should throw a more descriptive exception.

File: src/DotNetWorker/Configuration/WorkerOptions.cs [hunk 4]
```diff
@@ -1,29 +1,27 @@
...
+            {
+                PropertyNameCaseInsensitive = true
+            };
 
-            // Default value for HeartBeatRateFactor is 6
-            HeartBeatRateFactor = 6;
+            Serializer = new JsonObjectSerializer(serializerOptions);
```
Comment (fabiocav) [inline src/DotNetWorker/Configuration/WorkerOptions.cs:18]: Should this be deferred? If a different serializer is used, we shouldn't need to create this instance, right?

File: src/DotNetWorker/Configuration/WorkerOptions.cs (hunk 2 above)
Comment (brettsam) [reply src/DotNetWorker/Configuration/WorkerOptions.cs:13]: I debated this one -- I don't think anything internally matters here. But I was trying to pick the biggest difference between Json.NET and System.Text.Json. In retrospect, it makes sense (and it's easier to explain) if we just use the default here.

File: src/DotNetWorker/Configuration/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (fabiocav) [reply src/DotNetWorker/Configuration/ServiceCollectionExtensions.cs:78]: Assuming this didn't spark joy...

File: src/DotNetWorker/Configuration/WorkerOptions.cs [hunk 5]
```diff
@@ -1,29 +1,23 @@
...
-            // Default value for HeartBeatRateFactor is 6
-            HeartBeatRateFactor = 6;
+        /// <summary>
+        /// The <see cref="ObjectSerializer"/> to use for all JSON serialization and deserialization. By default,
+        /// this is a default <see cref="JsonObjectSerializer"/> with default <see cref="JsonSerializerOptions"/>.
+        /// </summary>
+        public ObjectSerializer Serializer
+        {
+            get => _serializer ??= new JsonObjectSerializer();
+            set => _serializer = value;
```
Comment (fabiocav) [inline src/DotNetWorker/Configuration/WorkerOptions.cs:20]: Should we perform a `null` check on the value? Do we ever want to allow this to be set to `null`?

**Review by fabiocav — APPROVED** [review]
Two small comments here, but looks good, so flagging as approved.

**Pull request merged**